File: core/src/com/ovt/quest/screens/MainMenuScreen.java
package com.ovt.quest.screens;


import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Screen;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.Stage;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import com.ovt.quest.QuestGame;

/**
 * Created by kombarov_na on 21.12.2017.
 */

public class MainMenuScreen implements Screen {

    QuestGame game;

    Stage stage = new Stage();

    public MainMenuScreen(QuestGame game) {
        this.game = game;
    }

    @Override
    public void show() {
        Gdx.input.setInputProcessor(stage);

        stage.addActor(game.background);

        game.switchFontBug(game.bigFont);

        TextButton playButton = new TextButton("Играть", game.biggerTextButtonStyle);
        playButton.addListener(toQuestScreen);

        TextButton exitButton = new TextButton("Выход", game.biggerTextButtonStyle);
        exitButton.addListener(exit);

        Table table = new Table();
        table.setFillParent(true);
        table.defaults().width(Gdx.graphics.getWidth() * 0.50f);

        table.add(playButton);
        table.row();
        table.add(exitButton);

        stage.addActor(table);
    }

    private ClickListener toQuestScreen = new ClickListener() {
        @Override
        public void touchUp(InputEvent event, float x, float y, int pointer, int button) {
            game.setScreen(new QuestScreen(game));
        }
    };

    private ClickListener exit = new ClickListener() {
        @Override
        public void touchUp(InputEvent event, float x, float y, int pointer, int button) {
            Gdx.app.exit();
        }
    };

    @Override
    public void render(float delta) {
        game.batch.begin();
        stage.draw();
        game.bigFont.draw(game.batch, "asdjfkl; фываолдж", 150, 150);
        game.batch.end();
    }

    @Override
    public void resize(int width, int height) { }

    @Override
    public void pause() { }

    @Override
    public void resume() { }

    @Override
    public void hide() { }

    @Override
    public void dispose() { }

}

File: core/src/com/ovt/quest/main_menu_screens/MainMenuScreen.kt
package com.ovt.quest.main_menu_screens


import com.badlogic.gdx.Gdx
import com.badlogic.gdx.Screen
import com.badlogic.gdx.graphics.GL20
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.scenes.scene2d.Stage
import com.badlogic.gdx.scenes.scene2d.ui.Image
import com.badlogic.gdx.scenes.scene2d.ui.Table
import com.ovt.quest.QuestGame
import com.ovt.quest.quest.QuestScreen

/**
 * Created by kombarov_na on 21.12.2017.
 */
class MainMenuScreen(internal var game: QuestGame) : Screen {

    private val stage = Stage()

    val t = Texture(Gdx.files.internal("img/фон-главное-меню.png"))
    val bg = Image(t)

    init {
        bg.x = 0f
        bg.y = 0f
        bg.width = Gdx.graphics.width.toFloat()
        bg.height = Gdx.graphics.height.toFloat()
    }

    override fun show() {

        val h = Gdx.graphics.height
        val w = Gdx.graphics.width

        Gdx.input.inputProcessor = stage

        val startButton = game.buttons.biggerButton("Начать", onClick = {
            clearGameProgress()
            game.screen = QuestScreen(game)
        })
        val continueButton = game.buttons.biggerButton("Продолжить", onClick = { game.screen = QuestScreen(game) })

        val minigamesButton = game.buttons.biggerButton("Миниигры", onClick = { game.screen = MinigamesScreen(game) })

        val exitButton = game.buttons.biggerButton("Выход", onClick = { Gdx.app.exit() })

        val table = Table()
        table.setFillParent(true)
        table.defaults().width(Gdx.graphics.width * 0.8f).pad(h * 0.008f)

        table.add(startButton)
        table.row()
        table.add(continueButton)
        table.row()
        table.add(minigamesButton)
        table.row()
//        table.add(exitButton)

        stage.addActor(bg)
        stage.addActor(table)
    }

    private fun clearGameProgress() {
        game.globals.currentQuestNode = game.globals.defaultQuestNode
        game.globals.questNodes.forEach { it.value.hidden = false }
    }

    override fun render(delta: Float) {
        Gdx.gl.glClearColor(1f, 1f, 1f, 1f)
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT)

        stage.draw()
    }

    override fun resize(width: Int, height: Int) {}

    override fun pause() {}

    override fun resume() {}

    override fun hide() {}

    override fun dispose() {
        t.dispose()
    }

}
